includes/Application.h

```diff
@@ -0,0 +1,117 @@
+#pragma once
+#include "FPSManager.h"
+#include "GameState.h"
+#include "GameStateManager.h"
+#include "WindowManager.h"
+#include <memory.h>
+
+namespace doengine
+{
+
+class FpsManager;
+class GameStateManager;
+class WindowManager;
+class Renderer;
+
+class ClassIdentifier
+{
+  public:
+    virtual std::string getClassIdentifier() = 0;
```

Comment:
I think it would be better to return the string as reference and by argument:
```suggestion
    virtual void getClassIdentifier(const std::string& classId) = 0;
```